src/MIDebugEngine/Engine.Impl/Disassembly.cs

```diff
@@ -166,10 +143,123 @@ public Disassembly(DebuggedProcess process)
                 }
                 var kv = block.TryFetch(address, nInstructions, out ret);
             }
-
             return ret;
         }
 
+        /// <summary>
+        /// Seek backward n instructions from a target address
+        /// </summary>
+        /// <param name="address">target address</param>
+        /// <param name="nInstructions">number of instructions</param>
+        /// <returns> address - n on failure, else the address of an instruction n back from the target address</returns>
+        public async Task<ulong> SeekBack(ulong address, int nInstructions)
+        {
+            ICollection<DisasmInstruction> ret = null;
+            ulong defaultAddr = address >= (ulong)nInstructions ? address - (ulong)nInstructions : 0;
+
+            lock (_disassemlyCache)
+            {
+                // check the cache
+                var kv = _disassemlyCache.FirstOrDefault((p) => p.Value.TryFetch(address, -nInstructions, out ret));
+                if (kv.Value != null)
+                    return ret.First().Addr;
+            }
+            ulong endAddress;
+            ulong startAddress;
+            var range = await _process.FindValidMemoryRange(address, (uint)(_process.MaxInstructionSize * nInstructions)+1, (int)(_process.MaxInstructionSize * -nInstructions));
```

Comment:
Since we only went back a single byte per instruction on line 158, shouldn't this value just be the starting value of `address`?